### ISSUE TITLE
Question about the setting of `lr_config` in `config.py`

### ISSUE THREAD

**Issue by zhongqiu1245** [open-mmlab/mmdetection#5436]
Hi, dear authors!
Sorry to bother you.
I have a question about the setting of `lr_config` in `config.py`
I see some config files have
`warmup='linear, warmup_iters=500, warmup_ratio=0.001, step=[16, 19]`
What do `warmup_iters` and `warmup_ratios` mean?
And how do I get the value of `step`? And how many `step` are set? Is there any empirical formula to follow?
Thank you!

**Comment by RangiLyu** [comment]
Here is a brief introduction of ld_config:
```
lr_config = dict(  # Learning rate scheduler config used to register LrUpdater hook
    policy='step',  # The policy of scheduler, also support CosineAnnealing, Cyclic, etc. Refer to details of supported LrUpdater from https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/lr_updater.py#L9.
    warmup='linear',  # The warmup policy, also support `exp` and `constant`.
    warmup_iters=500,  # The number of iterations for warmup
    warmup_ratio=
    0.001,  # The ratio of the starting learning rate used for warmup
    step=[8, 11])  # Steps to decay the learning rate
```
warmup means use a small lr to warmup the training process which can make training more stable. And the step means the in which epoch the learning rate will decay when using Step Learning Rate Scheduler.

**Comment by zhongqiu1245** [comment]
Hi @RangiLyu 
Does it mean `warmup_iters * warmup_ratio = lr` ?

**Comment by RangiLyu** [comment]
If you use linear warmup policy, it means the training learning rate will start with `warmup_ratio*lr` and then linearly increase to the `lr` set in optimizer after `warmup_iters` steps. These values are hyperparameters which means you need to adjust them according to different models and datasets.

**Comment by zhongqiu1245** [comment]
According to your opinion, `warmup_ratio` is only used once at the starting position of warmup process, and then the linear increase ratio in `warmup_iters` is actually `(lr-warmup_ratio*lr)/ warmup_iters` instead of `warmup_ratio` set at the beginning
Am I right?

**Comment by RangiLyu** [comment]
Yes, you are right.

**Comment by zhongqiu1245** [comment]
If I want to cancel `warm_up`, how do I set it up?

**Comment by RangiLyu** [comment]
You can set `warmup=None`.

**Comment by zhongqiu1245** [comment]
@RangiLyu Thank you
Is the step of `lr_config` an epoch? `step=[8, 11]` is to decay the learning rate once at epoch=8, and then decay it again at epoch=11

**Comment by RangiLyu** [comment]
> @RangiLyu Thank you
> Does the `step` of `lr_config` mean `epoch`?
> `step=[8, 11]` is to decay the learning rate once at epoch=8, and then decay it again at epoch=11?

You are right.

**Comment by zhongqiu1245** [comment]
thank you！

**Comment by TuanNguyenSKKU** [comment]
Dear @RangiLyu and @zhongqiu1245 
How can I choose the step values to optimize the training model? For example, Steps 8 and 11.

I hope to receive your response. Thank you.

**Comment by zhongqiu1245** [comment]
@TuanNguyenSKKU Hi
You can set step=[8, 11] in your condig.py

**Comment by TuanNguyenSKKU** [comment]
Thank you @zhongqiu1245 
But I mean what criteria to choose 8 and 11 instead of other values?

**Comment by FRAOTIAC** [comment]
Hi @zhongqiu1245, Is there an easy way to plot an lr curve based on lr_config?

**Comment by zhongqiu1245** [comment]
@TuanNguyenSKKU Hi
I think it should be selected by experience

**Comment by zhongqiu1245** [comment]
@FRAOTIAC Hi
you can add this code in your `_base_/default_runtime.py`.
code:`dict(type='TensorboardLoggerHook')`
like this:
`log_config = dict(
    interval=50,
    hooks=[
        dict(type='TextLoggerHook'),
        dict(type='TensorboardLoggerHook')   # tensorboard
    ])`
then you can view your lr_curve in your tensorboard.

**Comment by FRAOTIAC** [comment]
Thanks for the info, But I just want to plot the curve without training, to verify that the lr_config meets my expectations. Is this possible?

**Comment by FRAOTIAC** [comment]
Hi @zhongqiu1245 
Thanks for the info, But I just want to plot the curve without training, to verify that the lr_config meets my expectations. Is this possible?
